lib/cfg.c

```diff
@@ -620,6 +620,8 @@ cfg_free(GlobalConfig *self)
   if (self->original_config)
     g_string_free(self->original_config, TRUE);
 
+  g_list_free_full(self->file_list, g_free);
```

Comment:
`g_free` only frees `CfgFilePath`  area. The buffers allocated for the fields are not freed. You need to implement a new destructor for `CfgFilePath`, that needs to be called here.

```
static
cfg_file_path_free(CfgFilePath *self)
{
  g_free(self->file_path);
  g_free(self->file_type);
  g_free(self);
}
```
and
```
 g_list_free_full(self->file_list, cfg_file_path_free);
```

Reply:
@furiel : Yes, I have made all changes now. Thanks for useful feedback.